Firestore/core/test/firebase/firestore/model/field_value_test.cc

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/model/field_value.h"
+
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+using Type = FieldValue::Type;
+
+TEST(FieldValue, NullType) {
+  const FieldValue value = FieldValue::NullValue();
+  EXPECT_EQ(Type::Null, value.type());
+  EXPECT_FALSE(value < value);
+}
+
+TEST(FieldValue, BooleanType) {
+  const FieldValue true_value = FieldValue::BooleanValue(true);
+  const FieldValue false_value = FieldValue::BooleanValue(false);
+  EXPECT_EQ(Type::Boolean, true_value.type());
+  EXPECT_FALSE(true_value < true_value);
+  EXPECT_FALSE(true_value < false_value);
+  EXPECT_FALSE(false_value < false_value);
+  EXPECT_TRUE(false_value < true_value);
+}
+
+TEST(FieldValue, ArrayType) {
+  const FieldValue empty =
+      FieldValue::ArrayValue(std::vector<const FieldValue>{});
+  std::vector<const FieldValue> small_value_array = {
```

Comment:
I don't understand what the names `small_value_array` and `large_value_array` mean here. The small one seems longer than the large one.

Also, you can skip the assignment when initializing.

```c++
std::vector<const FieldValue> small_value_array{
  // ...
};
```

Reply:
Done